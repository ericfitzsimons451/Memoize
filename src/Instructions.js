import React from 'react';
import './main.scss'

function Instructions(props) {
  return(
    <article className="instructions-display">
            <h2 className="instructions-header">How to Use This App</h2>
            <p className="instructions-article">Click on any of the cards to display the correct answer.</p>
            <p className="instructions-article">After viewing the correct answer, click the button to return to the main display.</p>
            <button type="button" className="exit-instructions-button" onClick={props.returnToMainDisplay}>Let's Go!</button>
          </article>
    )
}

export default Instructions;