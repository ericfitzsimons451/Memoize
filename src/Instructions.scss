.instructions-display {
  position: absolute;
  top: 15px;
  z-index: 50;
  height: 700px;
  width: 800px;
  border-radius: 50px;
  background-color: white;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
}

.instructions-header {
  font-size: 3rem;
  background-color: green;
  margin: 50px 0px 0px 20px;
  width: 50%;
  height: 14%;
  border-radius: 20px;
}

.instructions-article {
  font-size: 2rem;
  margin: 5px;
}

.exit-instructions-button {
  width: 300px;
  height: 125px;
  margin: 10px;
  font-size: 1.8rem;
  background-color: yellow;
  border-radius: 15px;
}