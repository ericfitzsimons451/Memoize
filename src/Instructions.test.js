import React from 'react';
import Instructions from './Instructions.js'
import { shallow } from 'enzyme';

const mockReturnToMainDisplay = jest.fn();


describe('Instructions', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Instructions returnToMainDisplay={mockReturnToMainDisplay} />
      )
  })

  it('should match the snapshot with all data correctly passed to the Instructions', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should register a click when the LETs Go! button is pressed', () => {
    wrapper.find('.exit-instructions-button').simulate('click')
    expect(mockReturnToMainDisplay).toBeCalled()
  })
})