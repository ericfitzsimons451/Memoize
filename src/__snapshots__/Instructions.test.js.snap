// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions should match the snapshot with all data correctly passed to the Instructions 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Instructions
    returnToMainDisplay={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="instructions-header"
        >
          How to Use This App
        </h2>,
        <p
          className="instructions-article"
        >
          Click on any of the cards to display the correct answer.
        </p>,
        <p
          className="instructions-article"
        >
          After viewing the correct answer, click the button to return to the main display.
        </p>,
        <button
          className="exit-instructions-button"
          onClick={[MockFunction]}
          type="button"
        >
          Let's Go!
        </button>,
      ],
      "className": "instructions-display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "How to Use This App",
          "className": "instructions-header",
        },
        "ref": null,
        "rendered": "How to Use This App",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click on any of the cards to display the correct answer.",
          "className": "instructions-article",
        },
        "ref": null,
        "rendered": "Click on any of the cards to display the correct answer.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "After viewing the correct answer, click the button to return to the main display.",
          "className": "instructions-article",
        },
        "ref": null,
        "rendered": "After viewing the correct answer, click the button to return to the main display.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Let's Go!",
          "className": "exit-instructions-button",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": "Let's Go!",
        "type": "button",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="instructions-header"
          >
            How to Use This App
          </h2>,
          <p
            className="instructions-article"
          >
            Click on any of the cards to display the correct answer.
          </p>,
          <p
            className="instructions-article"
          >
            After viewing the correct answer, click the button to return to the main display.
          </p>,
          <button
            className="exit-instructions-button"
            onClick={[MockFunction]}
            type="button"
          >
            Let's Go!
          </button>,
        ],
        "className": "instructions-display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "How to Use This App",
            "className": "instructions-header",
          },
          "ref": null,
          "rendered": "How to Use This App",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click on any of the cards to display the correct answer.",
            "className": "instructions-article",
          },
          "ref": null,
          "rendered": "Click on any of the cards to display the correct answer.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "After viewing the correct answer, click the button to return to the main display.",
            "className": "instructions-article",
          },
          "ref": null,
          "rendered": "After viewing the correct answer, click the button to return to the main display.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Let's Go!",
            "className": "exit-instructions-button",
            "onClick": [MockFunction],
            "type": "button",
          },
          "ref": null,
          "rendered": "Let's Go!",
          "type": "button",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
